virtual_rainforest/models/soil/constants.py

```diff
@@ -50,3 +55,54 @@ class TempScalar:
     """Unclear exactly what this parameter is [units unclear]"""
     ref_temp: float = 30.0
     """Reference temperature [degrees C]"""
+
+
+MICROBIAL_TURNOVER_RATE = 0.036
+"""Microbial turnover rate [day^-1], this isn't a constant but often treated as one."""
+
+MAX_UPTAKE_RATE_LABILE_C = 0.35
+"""Maximum (theoretical) rate at which microbes can take up labile carbon [day^-1]."""
+
+NECROMASS_ADSORPTION_RATE = 0.025
+"""Rate at which necromass is adsorbed by soil minerals [day^-1].
+
+Taken from :cite:t:`abramoff_millennial_2018`, where it was obtained by calibration.
+"""
+
+HALF_SAT_MICROBIAL_ACTIVITY = 0.0072
+"""Half saturation constant for microbial activity (with increasing biomass)[kg C m^-2].
+"""
+
+LEACHING_RATE_LABILE_CARBON = 1.5e-3
+"""Leaching rate for labile carbon (lmwc) [day^-1]."""
+
+CARBON_INPUT_TO_POM = 2.0 / 3.0
+"""Proportion of carbon input that becomes particulate organic matter (POM) [unitless].
+
+Taken from :cite:t:`abramoff_millennial_2018`, this is justified there based on previous
+empirical work. However, this is something we will definitely completely alter down the
+line so no need to worry too much about references.
+"""
+
+LITTER_INPUT_RATE = 0.172 / 365.25
+"""Rate of litter input to the system [kg C m^-2 day^-1].
+
+This definitely is not a constant for our purposes. However,
+:cite:t:`abramoff_millennial_2018` use a constant litter input rate, so we shall also
+use one initially.
+"""
+
+
+@dataclass
```

Comment:
I only learned about this recently, but apparently you can make `dataclass`es immutable, which I think would make sense here:

```suggestion
@dataclass(frozen=True)
```

Reply:
Yup - I did the same thing over in `pyrealm` in a fit of paranoia to stop people altering constants mid-run!

Reply:
Ah yeah that definitely seems sensible!

Reply:
This might be a wider issue to apply to other constants classes?

Reply:
Have switched to using `@dataclass(frozen=True)`. I've also used [`typing.Final`](https://docs.python.org/3/library/typing.html#typing.Final) for the constants that aren't stored in `data` classes, this won't stop them from being altered but will cause a `mypy` error if they are altered